### PR TITLE
Add ability to specify report headers

### PR DESCRIPTION
**Pull request by oscarwyatt**
Only some headers are required for FCDO report
Also need the ability to get the total number
of subscribers

Trello: https://trello.com/c/ipkKc0gg/751-to-automate-the-rake-tasks-associated-with-updating-fcdo-living-in-guides-eu-countries-and-travel-advice

### CODE REVIEW

File: docs/analytics.md [hunk 1]
```diff
@@ -23,6 +23,12 @@ TAGS_PATTERN=brexit_checklist_criteria report:csv_subscriber_lists["2020-11-05"]
 LINKS_PATTERN=countries report:csv_subscriber_lists["2020-11-05"]
 ```
 
+If you want just some specific headers, you can choose which headers you want as follows:
+```bash
+HEADERS=title,individual_subscribers,daily_subscribers,weekly_subscribers,total_subscribers report:csv_subscriber_lists["2020-11-05"]
+```
+All available headers are used by default, simply pass in the name of the headers as they are in the first row of the default output
```
Comment (chao-xian) [inline docs/analytics.md:30]: Nitpick but please remove `simply`.

Comment (chao-xian) [reply docs/analytics.md:30]: Can you also add an example below on how to run the rake task equivalent please?

File: lib/reports/subscriber_lists_report_row.rb [hunk 2]
```diff
@@ -0,0 +1,81 @@
+class Reports::SubscriberListsReportRow
+  def initialize(date, headers, list)
+    @date = date
+    @headers = headers
+    @list = list
+  end
+
+  def call
+    headers.map { |header| send(header) }
```
Comment (chao-xian) [inline lib/reports/subscriber_lists_report_row.rb:9]: What's the advantage of doing this versus:

```ruby

	headers.map { |header| &:header }
```

(...apart from being confusingly named)?

Comment (oscarwyatt) [reply lib/reports/subscriber_lists_report_row.rb:9]: I'm not sure I understand your question...it's calling send on each member of the array `headers` - ie so headers might be `[:title, :slug]` - line 9 simply iterates over the array and calls the relevant function by passing the member of the array to the `send` function, which calls the relevant function (ie for the element `:title` the `title` method of the class will get called. AFAIK calling `&:header` will just try to call the header function for every member of the array - or have I misunderstood something?

File: docs/analytics.md (hunk 1 above)
Comment (oscarwyatt) [reply docs/analytics.md]: 👍

File: lib/reports/subscriber_lists_report_row.rb (hunk 2 above)
Comment (chao-xian) [reply lib/reports/subscriber_lists_report_row.rb:9]: Ahhhh sorry I didn't grok this properly. Thanks for the explainer. And validate headers stops the report entirely?

Comment (oscarwyatt) [reply lib/reports/subscriber_lists_report_row.rb:9]: Yup. I figured if someone was entered the wrong name by accident it's better to tell them than not include that header (which might lead to a lot of head scratching)

**Review by chao-xian — APPROVED** [review]

**Pull request merged**